opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -1,61 +1,129 @@
-use opentelemetry::logs::{LogRecord, Logger, LoggerProvider, Severity};
+use opentelemetry::{
+    logs::{AnyValue, LogRecord, Logger, LoggerProvider, Severity},
+    Key,
+};
 use std::borrow::Cow;
-use tracing_core::Level;
+use tracing_core::{Level, Metadata};
+use tracing_log::NormalizeEvent;
 use tracing_subscriber::Layer;
 
 const INSTRUMENTATION_LIBRARY_NAME: &str = "opentelemetry-appender-tracing";
 
 /// Visitor to record the fields from the event record.
-struct EventVisitor<'a> {
-    log_record: &'a mut LogRecord,
+#[derive(Default)]
+struct EventVisitor {
+    log_record_attributes: Vec<(Key, AnyValue)>,
+    log_record_body: Option<AnyValue>,
 }
 
-impl<'a> tracing::field::Visit for EventVisitor<'a> {
+#[cfg(feature = "experimental_metadata_attributes")]
+fn is_metadata(field: &'static str) -> bool {
```

Comment:
I am not able to understand the need for `is_metadata` check. Can you help explain the need for this?

Reply:
Unfortunately, when I have logs from the log crate, some metadata are in double.

So, if I modify the is_metadata function to always returns false, I have this :

````
otel-collector-1  | Body: Str(testlog)
otel-collector-1  | Attributes:
otel-collector-1  |      -> name: Str(log event)
otel-collector-1  |      -> log.target: Str(testotel)
otel-collector-1  |      -> log.module.path: Str(testotel)
otel-collector-1  |      -> log.source.file.path: Str(src/main.rs)
otel-collector-1  |      -> log.source.file.name: Str(main.rs)
otel-collector-1  |      -> log.source.file.line: Int(25)
otel-collector-1  |      -> log.target: Str(testotel)
otel-collector-1  |      -> log.module_path: Str(testotel)
otel-collector-1  |      -> log.file: Str(src/main.rs)
otel-collector-1  |      -> log.line: Str(25)
````

Reply:
Here is why it is like that. https://docs.rs/tracing-log/latest/tracing_log/trait.NormalizeEvent.html